### PR TITLE
Make code compatible with mypy 0.990

### PR DESCRIPTION
**Pull request by robsdedude**
* remove implicit optional parameters (PEP 484 received an update)
 * mute metaclass inference limitations of mypy
 * fix mypy rightfully complaining about empty method bodies
 * fixed bug in (deprecated) `session.last_bookmark` that mypy uncovered

### CODE REVIEW

File: neo4j/_async/work/session.py [hunk 1]
```diff
@@ -321,7 +323,7 @@ async def last_bookmark(self) -> t.Optional[str]:
         if self._auto_result:
             await self._auto_result.consume()
 
-        if self._transaction and self._transaction._closed:
+        if self._transaction and self._transaction._closed():
```
Comment (thelonelyvulpes) [inline neo4j/_async/work/session.py:326]: this seems unrelated to mypy

**Review by bigmontz — APPROVED** [review]
🥇 

(Quite nice bug solved 🎁 )

File: neo4j/_async/work/session.py (hunk 1 above)
Comment (bigmontz) [reply neo4j/_async/work/session.py:326]: the linter may point out `closed` is a function, not a property.

**Review by thelonelyvulpes — APPROVED** [review]
